File: backend/schemas/job.py
from pydantic import BaseModel
from uuid import UUID
from datetime import datetime


class JobStatusResponse(BaseModel):
    id: UUID
    status: str
    progress: int
    error_message: str | None
    started_at: datetime | None
    completed_at: datetime | None
    created_at: datetime

    class Config:
        from_attributes = True


class TrainingJobResponse(JobStatusResponse):
    voice_profile_id: UUID
    epochs: int
    current_epoch: int
    loss: float | None


class GenerationJobResponse(JobStatusResponse):
    job_type: str
    voice_profile_id: UUID
    output_audio_url: str | None
    duration_seconds: float | None


class JobListResponse(BaseModel):
    jobs: list[JobStatusResponse]
    total: int

File: frontend/src/types/job.ts
export type JobStatus = "queued" | "processing" | "completed" | "failed" | "cancelled";
export type JobType = "training" | "tts" | "conversion";

export interface Job {
  id: string;
  status: JobStatus;
  progress: number;
  error_message: string | null;
  started_at: string | null;
  completed_at: string | null;
  created_at: string;
}

export interface TrainingJob extends Job {
  voice_profile_id: string;
  epochs: number;
  current_epoch: number;
  loss: number | null;
}

export interface GenerationJob extends Job {
  job_type: JobType;
  voice_profile_id: string;
  output_audio_url: string | null;
  duration_seconds: number | null;
}

export interface JobListResponse {
  jobs: Job[];
  total: number;
}
